pkg/controller/job/job_controller.go

```diff
@@ -846,6 +840,41 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 			jobCtx.podsWithDelayedDeletionPerIndex = getPodsWithDelayedDeletionPerIndex(logger, jobCtx)
 		}
 	}
+
+	complete := false
+	if jobCtx.finishedCondition == nil {
+		if job.Spec.Completions == nil {
+			// This type of job is complete when any pod exits with success.
+			// Each pod is capable of
+			// determining whether or not the entire Job is done.  Subsequent pods are
+			// not expected to fail, but if they do, the failure is ignored.  Once any
+			// pod succeeds, the controller waits for remaining pods to finish, and
+			// then the job is complete.
+			complete = jobCtx.succeeded > 0 && active == 0
+		} else {
+			// If job is scaled down and the number of succeeded pods already reached completions,
+			// job should be marked as complete here.
+			wantActive := active
+			if feature.DefaultFeatureGate.Enabled(features.ElasticIndexedJob) && satisfiedExpectations && job.DeletionTimestamp == nil {
```

Comment:
Do we need this `if`? Maybe, if needed, it could be inside the `wantActivePods` function?

Reply:
I think we should put this `if` here since `wantActivePods` is also called in L1478.
@mimowo WDYT?

Reply:
@mimowo Friendly ping :)

Reply:
Ah, you're right. We should call inside `wantActive()`. Thanks!